### PR TITLE
Added api_version field in every record

### PR DESCRIPTION
**Pull request by whiteshadow-gr**
Added one extra field, `api_version`. It will be used on the SHE feed to support both old and new instagram items

### CODE REVIEW

File: dataplug-instagram/app/com/hubofallthings/dataplugInstagram/apiInterfaces/InstagramFeedInterface.scala [hunk 1]
```diff
@@ -146,6 +152,24 @@ class InstagramFeedInterface @Inject() (
...
 
+  private def transformData(rawData: JsValue): JsResult[JsObject] = {
+    val transformation = (__ \ "data").json.update(
+      __.read[JsArray].map { feedData =>
+        {
+          val updatedFeedData = feedData.value.map { item =>
+            val unixTimeStamp = DateTime.parse((item \ "timestamp").asOpt[String].getOrElse("")).getMillis / 1000
+            item.as[JsObject] ++ JsObject(Map(
+              "api_version" -> JsString("v2"),
+              "created_time" -> JsString(unixTimeStamp.toString)))
```
Comment (augustinas) [inline dataplug-instagram/app/com/hubofallthings/dataplugInstagram/apiInterfaces/InstagramFeedInterface.scala:163]: Both fields should start with a `ds` prefix to clearly indicate organisation adding these fields into the record. Prefix should be configurable through the environment variables with a default for `ds`.

File: dataplug-instagram/app/com/hubofallthings/dataplugInstagram/apiInterfaces/InstagramProfileInterface.scala [hunk 2]
```diff
@@ -78,12 +78,11 @@ class InstagramProfileInterface @Inject() (
   }
 
   private def transformData(rawData: JsValue): JsResult[JsObject] = {
-
     val transformation = __.json.update(
       __.read[JsObject].map(profile => {
-
         profile ++ JsObject(Map(
-          "hat_created_time" -> JsString(LocalDateTime.now().toString)))
+          "hat_created_time" -> JsString(LocalDateTime.now().toString),
```
Comment (augustinas) [inline dataplug-instagram/app/com/hubofallthings/dataplugInstagram/apiInterfaces/InstagramProfileInterface.scala:84]: Please update organisation prefix `hat` -> `ds`

File: dataplug-instagram/app/com/hubofallthings/dataplugInstagram/apiInterfaces/InstagramProfileInterface.scala [hunk 3]
```diff
@@ -78,12 +78,11 @@ class InstagramProfileInterface @Inject() (
...
+          "api_version" -> JsString("v2")))
```
Comment (augustinas) [inline dataplug-instagram/app/com/hubofallthings/dataplugInstagram/apiInterfaces/InstagramProfileInterface.scala:85]: Add organisation prefix

File: dataplug-instagram/app/com/hubofallthings/dataplugInstagram/apiInterfaces/InstagramFeedInterface.scala (hunk 1 above)
Comment (whiteshadow-gr) [reply dataplug-instagram/app/com/hubofallthings/dataplugInstagram/apiInterfaces/InstagramFeedInterface.scala:163]: Sure no problem. Why should it be configurable? What's the use case?

**Comment by whiteshadow-gr** [comment]
FYI: Squashed the 2 last commits together.

File: dataplug-instagram/app/com/hubofallthings/dataplugInstagram/apiInterfaces/InstagramFeedInterface.scala [hunk 4]
```diff
@@ -146,6 +153,25 @@ class InstagramFeedInterface @Inject() (
...
+  private def transformData(rawData: JsValue): JsResult[JsObject] = {
+    val prependFieldsId = configuration.get[String]("service.customFieldId")
+    val transformation = (__ \ "data").json.update(
+      __.read[JsArray].map { feedData =>
+        {
+          val updatedFeedData = feedData.value.map { item =>
+            val unixTimeStamp = DateTime.parse((item \ "timestamp").asOpt[String].getOrElse("")).getMillis / 1000
+            item.as[JsObject] ++ JsObject(Map(
+              s"${prependFieldsId}_api_version" -> JsString("v2"),
+              "created_time" -> JsString(unixTimeStamp.toString)))
```
Comment (whiteshadow-gr) [inline dataplug-instagram/app/com/hubofallthings/dataplugInstagram/apiInterfaces/InstagramFeedInterface.scala:165]: This has to be the same as v1. Or I copy paste the instagram mapper on HAT and we have 2

File: dataplug-instagram/app/com/hubofallthings/dataplugInstagram/apiInterfaces/InstagramProfileInterface.scala [hunk 5]
```diff
@@ -78,12 +79,12 @@ class InstagramProfileInterface @Inject() (
...
 
   private def transformData(rawData: JsValue): JsResult[JsObject] = {
-
+    val prependFieldsId = configuration.get[String]("service.customFieldId")
     val transformation = __.json.update(
       __.read[JsObject].map(profile => {
-
         profile ++ JsObject(Map(
-          "hat_created_time" -> JsString(LocalDateTime.now().toString)))
+          "hat_created_time" -> JsString(LocalDateTime.now().toString),
```
Comment (whiteshadow-gr) [inline dataplug-instagram/app/com/hubofallthings/dataplugInstagram/apiInterfaces/InstagramProfileInterface.scala:86]: It has to be the same as v1. It is used for sorting the items

**Pull request merged**